wurst/util/Colors.wurst

```diff
@@ -60,13 +62,77 @@ public tuple color(int red, int green, int blue)
 /** Tuple for your RGBA color needs. 0 = black/transparent, 255 = full color/opaque */
 public tuple colorA(int red, int green, int blue, int alpha)
 
+/** Tuple for your HSV(Hue Saturation Value) color needs. All values 0-1 */
+public tuple colorHSV(real h, real s, real v)
+
 /** Add an alpha value to a simple color tuple */
 public function color.withAlpha(int a) returns colorA
 	return colorA(this.red, this.green, this.blue, a)
 
 public function colorA.withoutAlpha() returns color
 	return color(this.red, this.green, this.blue)
 
+/** Returns the HSV equivalent */ 
+public function color.toHSV() returns colorHSV
+	real s
+	real h
+	let r = this.red / 255.0
+	let g = this.green / 255.0
+	let b = this.blue / 255.0
+	let max = max(r, g, b)
+	let min = min(r, g, b)
+	let delta = max - min
+	if max != 0 and delta != 0
+		s = delta / max
+		if r == max
+			if g < b
+				h = 1 + (g - b) / delta / 6
+			else
+				h = (g - b) / delta / 6
+		else if g == max
+			h = 1.0 / 3 + (b - r) / delta / 6
+		else
+			h = 2.0 / 3 + (r - g) / delta / 6
+	else
+		h = 0
```

Comment:
Why aren't s and h initialized with these values?

Reply:
If they are, they would be assigned twice if the max and delta are not 0
A small speed optimization, If you are against it, I will change it

Reply:
Yea, it makes the code uglier and longer, and I highly doubt that it would make a noticeable difference. I also think color transformations usually don't happen in performance critical situations?

Reply:
My opinion is otherwise, but sure, It's your library

Reply:
If you can provide a real world example where this matters (i.e. noticeable improvement), feel free to do so.
You found out yourself that we value readability and safety over these microop optimizations shenanigans.